### PR TITLE
Pending request and process request notifications

### PR DESCRIPTION
**Pull request by Sarahb-da**
This PR gets rid of "Requests" tables and just puts them in the notification center. Also introduces the concept of pending requests

Screen Shot 2021-06-24 at 12 15 37 PM
Screen Shot 2021-06-24 at 12 16 01 PM

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -320,7 +317,12 @@ const AppComponent = () => {
   });
 
   const notifications = useAllNotifications(party);
-  const notifCount = notifications.reduce((count, { contracts }) => count + contracts.length, 0);
+  const notifCount = notifications.reduce((count, { contracts, kind }) => {
+    if (kind === 'Pending') {
```
Comment (Sarahb-da) [inline ui/src/App.tsx:321]: dont show the green notification bubble for pending request notifications

File: ui/src/pages/notifications/Notifications.tsx [hunk 2]
```diff
@@ -706,11 +894,30 @@ const Notifications: React.FC<Props> = ({ notifications }) => {
...
+                      contract={c}
+                    />
+                  ));
                 default:
                   return null;
               }
             })
-          : 'No Notifications.'}
+          : 'No Notifications or Pending Requests.'}
+        <>{auctionCustomer.length > 0 && <AuctionRequests services={auctionService} />}</>
```
Comment (Sarahb-da) [inline ui/src/pages/notifications/Notifications.tsx:920]: will add auction requests in next PR

**Review by alexmatson-da — APPROVED** [review]
looks good!

File: ui/src/pages/common.ts [hunk 3]
```diff
@@ -54,3 +54,15 @@ export function damlSetValues<T>(damlSet: DamlSet<T>): T[] {
   }
   return r;
 }
+
+export function partitionArray<T>(isValid: (elem: T) => boolean, array?: T[]): [T[], T[]] {
```
Comment (brian-weir) [inline ui/src/pages/common.ts:58]: This looks identical to `_.partition` in the `lodash` library.

From the lodash docs at see https://lodash.com/docs/4.17.15#partition :

> Creates an array of elements split into two groups, the first of which contains elements predicate returns truthy for, the second of which contains elements predicate returns falsey for. The predicate is invoked with one argument: (value).

**Review by brian-weir — APPROVED** [review]
lgtm!

**Pull request merged**